packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -477,41 +526,30 @@ class Treemap extends StatelessWidget {
       preferBelow: false,
       child: InkWell(
         onTap: () {
-          if (rootNode.children.isNotEmpty) onRootChangedCallback(newRoot);
+          onRootChangedCallback(newRoot);
         },
         child: child,
       ),
     );
   }
 
-  Widget buildSubTreemaps() {
-    return LayoutBuilder(
-      builder: (context, constraints) {
-        // TODO(peterdjlee): Investigate why exception is thrown without this check
-        //                   and if there are any other cases.
-        if (constraints.maxHeight == 0 || constraints.maxWidth == 0) {
-          return const SizedBox();
-        }
-        final positionedChildren = buildTreemaps(
-          children: nodes,
-          x: 0,
-          y: 0,
-          width: constraints.maxWidth,
-          height: constraints.maxHeight,
-        );
-        if (levelsVisible <= 1) {
-          // If this is the second to the last level, paint all cells in the last level
-          // instead of creating widgets to improve performance.
-          return CustomPaint(
-            painter: MultiCellPainter(nodes: positionedChildren),
-            size: Size(constraints.maxWidth, constraints.maxHeight),
-          );
-        } else {
-          // Else all widgets should still be positioned Treemap widgets.
-          return Stack(children: positionedChildren);
-        }
-      },
-    );
+  /// Checks if the touch point of the given [details] is overlapping with
+  /// a cell in [positionedCells].
+  ///
+  /// If so, reroot to the matching cell if the cell has children to show.
+  void _onTapDown(
+    TapDownDetails details,
+    List<PositionedCell> positionedCells,
+  ) {
+    final x = details.localPosition.dx;
+    final y = details.localPosition.dy;
+    final touchPoint = Offset(x, y);
+    // TODO(peterdjlee): Optimize with better algorithm to find the overlapping cell.
```

Comment:
nit: better -> more efficient
just to be clear that this algorithm is correct and you are concerned about the case where positionedCells is large. You could say this is O(positionedCells.length) while an optimized algorithm could be O(log(positionCells.length) if you want to be a little clearer.